### ISSUE TITLE
Distinct exit code for warnings

### ISSUE THREAD

**Issue by gardar** [ansible/ansible-lint#3970]
##### Summary

I feel that there should be a distinction in exit codes between failures/warnings/success.

Currently it seems that ansible-lint has the exit code `2` when there is a failure but for warnings it has the exit code `0` which is the same as exit code as success and if strict mode (`-s`) is used then the warnings get the same exit code `2` as failures.
This overlap in exit codes between different outcomes (success, warnings, failures) presents a significant limitation for example when attempting to pass the outcome to a CI job status.

[Yamllint has the following exit codes](https://yamllint.readthedocs.io/en/stable/configuration.html#errors-and-warnings) which I feel would be appropriate for ansible-lint too

        0 if no errors or warnings occur
        1 if one or more errors occur
        2 if no errors occur, but one or more warnings occur

##### Issue Type

- Bug Report

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
@gardar Basically you want an unique exit code for the case where you run linter with `--strict` and it fails only with warnings?

Is that is all, it seems doable, even if I would be quite curious to see how you plan to use it in practice.

Im my experience, you either ignore warnings or fail on them on CI, based on project lead preference.

When not using strict mode, we cannot really do anything as we need to keep returning success (0).

**Comment by gardar** [comment]
Yes exactly! A different exit code for warnings and errors where the error exit code has precedence over the warning one.

In my case it will be used in Gitlab CI which I can configure to output a warning status based on a exit code, not sure if github actions support something similar.
```yaml
  allow_failure:
    exit_codes: 2
```

Now practically speaking then the idea is to bring attention to the warnings as it's recommended to fix the warnings but not a hard requirement.  
It's useful for example when you enable the linter in a repo with a lot of old code which you want want to fix eventually.
Sure you can ignore the warnings indefinitely, but the same could be said for warnings in monitoring systems or heck even for warnings in general.

Any number is fine although it would have been nice to use the same ones as yamllint. Is there any specific reason why ansible-lint uses exit code 2 by default as opposed to 1?

Adding another parameter to distinguish warnings from errors is also an option if you are afraid that changing the strict mode would be a breaking change.

**Comment by ssbarnea** [comment]
I am closing this as we have no plans to implement it but with the note that if someone will make a PR to implement this, we will review it. Start by looking at https://github.com/ansible/ansible-lint/blob/e469b1a338fedb997d3a953df3e8b45126661350/src/ansiblelint/app.py#L257 -- feel free to add an extra return code for `RC.FAILED_WITH_JUST_WARNINGS = 6` as other codes are used and we do not want to use RC = 1 (overlaps with command not found on shells).

**Comment by Joshua-rose** [comment]
A 2 line workaround for no warning return code. 
```bash
ansible-lint -p > ansible-lint.txt && if [ (wc -l ansible-lint.txt) -gt 0 ]; then exit <your exit code here>; fi
```
or 
```bash
ansible-lint 1> ansible-lint.txt && if [ (wc -l ansible-lint.txt) -gt 0 ]; then exit <your exit code here>; fi
```


additionally for context here are the return codes
```py
    SUCCESS = 0
    VIOLATIONS_FOUND = 2
    INVALID_CONFIG = 3
    LOCK_TIMEOUT = 4
    NO_FILES_MATCHED = 5
    EXIT_CONTROL_C = 130
```
